test/preview/components/routers/test_conditional_router.py

```diff
@@ -0,0 +1,305 @@
+        ]
+        router = ConditionalRouter(routes)
+        # enough_streams output slot will be selected with [1, 2, 3] list being outputted
+        kwargs = {"streams": [1, 2, 3], "query": "Haystack"}
+        result = router.run(**kwargs)
+        assert result == {"enough_streams": [1, 2, 3]}
+
+    def test_complex_condition(self):
+        routes = [
+            {
+                "condition": "{{messages[-1].metadata.finish_reason == 'function_call'}}",
+                "output": "{{streams}}",
+                "output_type": List[int],
+                "output_name": "streams",
+            },
+            {
+                "condition": "{{True}}",
+                "output": "{{query}}",
+                "output_type": str,
+                "output_name": "query",
+            },  # catch-all condition
+        ]
+        router = ConditionalRouter(routes)
+        message = mock.MagicMock()
+        message.metadata.finish_reason = "function_call"
+        result = router.run(messages=[message], streams=[1, 2, 3], query="my query")
+        assert result == {"streams": [1, 2, 3]}
+
+    def test_router_no_route(self, router):
+        # should raise an exception
+        router = ConditionalRouter(
+            [
+                {
+                    "condition": "{{streams|length < 2}}",
+                    "output": "{{query}}",
+                    "output_type": str,
+                    "output_name": "query",
+                },
+                {
+                    "condition": "{{streams|length >= 5}}",
+                    "output": "{{streams}}",
+                    "output_type": List[int],
+                    "output_name": "streams",
+                },
+            ]
+        )
+
+        kwargs = {"streams": [1, 2, 3], "query": "test"}
+        with pytest.raises(NoRouteSelectedException):
+            router.run(**kwargs)
+
+    def test_router_raises_value_error_if_route_not_dictionary(self):
+        """
+        Router raises a ValueError if each route is not a dictionary
+        """
+        routes = [
+            {"condition": "{{streams|length < 2}}", "output": "{{query}}", "output_type": str, "output_name": "query"},
+            ["{{streams|length >= 2}}", "streams", List[int]],
+        ]
+
+        with pytest.raises(ValueError):
+            ConditionalRouter(routes)
+
+    def test_router_raises_value_error_if_route_missing_keys(self):
+        """
+        Router raises a ValueError if each route does not contain 'condition', 'output', and 'output_type' keys
+        """
+        routes = [
+            {"condition": "{{streams|length < 2}}", "output": "{{query}}"},
+            {"condition": "{{streams|length < 2}}", "output_type": str},
+        ]
+
+        with pytest.raises(ValueError):
+            ConditionalRouter(routes)
+
+    def test_output_type_serialization(self):
+        assert serialize_type(str) == "builtins.str"
+        assert serialize_type(List[int]) == "typing.List"
```

Comment:
I'm afraid this won't be enough to deserialize it into a type that Canals can use for a connection. We definitely need to preserve the `int` as well.

Reply:
@ZanSara I can adjust code to serialize List[int] into `typing.List[int]` str, but what about deserialization? Is deserialization into typing.List enough?

Reply:
Ok, all should be covered now with [7eb0943](https://github.com/deepset-ai/haystack/pull/6147/commits/7eb09436d465767769900b7a4075aec9e934dfc4)